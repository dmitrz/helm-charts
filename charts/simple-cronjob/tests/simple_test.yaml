suite: simple-app simple case

values:
  - simple-values.yaml

release:
  name: simple
  namespace: default

chart:
  version: 0.0.0

tests:
  - it: All manifests should match snapshot
    templates:
      - cronjob.yaml
      - serviceaccount.yaml
      - pvc.yaml
    asserts:
      - matchSnapshot: {}

  - it: CronJob should be named as release
    template: cronjob.yaml
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: metadata.name
          value: simple

  - it: CronJob image tag should be set correctly
    template: cronjob.yaml
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].image
          value: test:1.1.1

  - it: CronJob should use imagePullSecrets if specified
    template: cronjob.yaml
    set:
      imagePullSecrets:
        - name: test
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.imagePullSecrets
          value:
            - name: test

  - it: CronJob should mount shared memory volume if shmMem specified
    template: cronjob.yaml
    set:
      shmSize: 1Gi
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[0]
          value:
            mountPath: /dev/shm
            name: shared-memory
      - equal:
          path: spec.jobTemplate.spec.template.spec.volumes[0]
          value:
            emptyDir:
              medium: Memory
              sizeLimit: 1Gi
            name: shared-memory

  - it: CronJob should have correctly mounted configs
    template: cronjob.yaml
    set:
      configs:
        - name: test.json
          mountPath: /etc/test.json
          value: |
            {
              "test": "test"
            }
        - name: test2.json
          value: |
            {
              "test2": "test2"
            }
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[1]
          value:
            mountPath: /etc/test.json
            name: configs
            subPath: test.json
            readOnly: true
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[2]
          value:
            mountPath: /configs/test2.json
            name: configs
            subPath: test2.json
            readOnly: true
      - contains:
          path: spec.jobTemplate.spec.template.spec.volumes
          content:
            configMap:
              name: simple
            name: configs

  - it: CronJob should have correctly mounted volumes with pvc
    template: cronjob.yaml
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[0]
          value:
            mountPath: /data
            name: data
      - equal:
          path: spec.jobTemplate.spec.template.spec.volumes[0]
          value:
            name: data
            persistentVolumeClaim:
              claimName: simple-data

  - it: CronJob should have correctly mounted volumes without pvc
    template: cronjob.yaml
    set:
      volumes:
        - name: data
          size: 1Gi
          mountPath: /data
          emptyDir: true
    asserts:
      - isKind:
          of: CronJob
      - equal:
          path: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[0]
          value:
            mountPath: /data
            name: data
      - equal:
          path: spec.jobTemplate.spec.template.spec.volumes[0]
          value:
            name: data
            emptyDir:
              sizeLimit: 1Gi

  - it: PVC should not be created if emptyDir
    template: pvc.yaml
    set:
      volumes:
        - name: data
          size: 1Gi
          mountPath: /data
          emptyDir: true
    asserts:
      - hasDocuments:
          count: 0

  - it: PVC should have correctly mounted volumes with pvc
    template: pvc.yaml
    asserts:
      - isKind:
          of: PersistentVolumeClaim
      - equal:
          path: spec.resources.requests.storage
          value: 1Gi

  - it: PVC should have correctly mounted volumes without pvc
    template: pvc.yaml
    set:
      volumes:
        - name: data
          size: 1Gi
          mountPath: /data
          emptyDir: true
    asserts:
      - hasDocuments:
          count: 0
