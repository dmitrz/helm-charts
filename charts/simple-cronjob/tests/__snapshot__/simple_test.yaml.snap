All manifests should match snapshot:
  1: |
    apiVersion: batch/v1
    kind: CronJob
    metadata:
      labels:
        app.kubernetes.io/instance: simple
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: simple-cronjob
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: simple-cronjob-0.0.0
      name: simple
    spec:
      concurrencyPolicy: Forbid
      jobTemplate:
        spec:
          template:
            metadata:
              labels:
                app.kubernetes.io/instance: simple
                app.kubernetes.io/name: simple-cronjob
            spec:
              containers:
                - env:
                    - name: TEST
                      value: test
                  image: test:1.1.1
                  imagePullPolicy: IfNotPresent
                  name: simple
                  resources: {}
                  securityContext: {}
                  volumeMounts:
                    - mountPath: /data
                      name: data
              restartPolicy: OnFailure
              securityContext: {}
              serviceAccountName: simple
              volumes:
                - name: data
                  persistentVolumeClaim:
                    claimName: simple-data
      schedule: '*/1 * * * *'
      timeZone: GMT
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app.kubernetes.io/instance: simple
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: simple-cronjob
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: simple-cronjob-0.0.0
      name: simple-data
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
  3: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/instance: simple
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: simple-cronjob
        app.kubernetes.io/version: 1.16.0
        helm.sh/chart: simple-cronjob-0.0.0
      name: simple
